File: app/src/main/java/com/kode/utair/data/network/NetworkCheckInterceptor.java
package com.kode.utair.data.network;

import com.kode.utair.domain.exceptions.NoNetworkException;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class NetworkCheckInterceptor implements Interceptor {

    private NetworkChecker networkChecker;

    public NetworkCheckInterceptor(NetworkChecker networkChecker) {
        this.networkChecker = networkChecker;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request.Builder requestBuilder = chain.request().newBuilder();

        if (!networkChecker.isConnected()) {
            throw new NoNetworkException("No network connection");
        }

        return chain.proceed(requestBuilder.build());
    }

}

File: app/src/main/java/com/kode/utair/data/network/NetworkCheckInterceptor.kt
package com.kode.utair.data.network

import com.kode.utair.domain.exceptions.NoNetworkException
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class NetworkCheckInterceptor(private val networkChecker: NetworkChecker) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val requestBuilder = chain.request().newBuilder()
        if (!networkChecker.isConnected) {
            throw NoNetworkException("No network connection")
        }
        return chain.proceed(requestBuilder.build())
    }

}